ugit

```diff
@@ -266,6 +267,29 @@ undo_all() {
     fi
 }
 
+undo_git_tag() {
+    # Fetch all tags from remote
+    if git fetch --all --tags > /dev/null 2>&1; then
+        printf "\n Fetching tags from remote"
+    else
+        perror "Unable to fetch tags from remote.Please check repository access."
+        exit
+    fi
+    tag=$(git tag | fzf --ansi --height 80% \
+        --reverse --multi --header="Choose tag to undo" \
+        --preview "echo {} | cut -d' ' -f1 | xargs -I{} git show --color --pretty=format:%b {}" \
+        --bind 'j:down,k:up,ctrl-j:preview-down,ctrl-k:preview-up,ctrl-space:toggle-preview' --preview-window right:60% )
+
+    # Exit if no tag is selected
+    [ -z "$tag" ] && exit
+
+    # Delete git tag
+    printf "%s\nDeleting tag: $tag"
+    git push origin --delete "$tag" > /dev/null 2>&1
+    git tag -d "$tag" > /dev/null 2>&1
```

Comment:
Add both these statements in `if` safe check. 
1. Remove it locally, if it succeds only then remove it from Remote